### ISSUE TITLE
samba update issue

### ISSUE THREAD

**Issue by kozistan** [fumatchu/RADS#5]
HI, with dnf update new samba is avaliable:

samba                                                          x86_64                              4.20.2-2.el9_5     
samba-client                                               x86_64                              4.20.2-2.el9_5                       
samba-client-libs                                        x86_64                              4.20.2-2.el9_5                                
samba-common                                           noarch                              4.20.2-2.el9_5                                 
samba-common-libs                                  x86_64                              4.20.2-2.el9_5                               
     replacing  samba-dc.x86_64 4.19.4-105.el9_4.dc
samba-common-tools                                 x86_64                              4.20.2-2.el9_5                                     
samba-dc-libs                                              x86_64                              4.20.2-2.el9_5                                          
samba-dcerpc                                              x86_64                              4.20.2-2.el9_5                                           
samba-ldb-ldap-modules                           x86_64                              4.20.2-2.el9_5                                          
samba-libs                                                    x86_64                              4.20.2-2.el9_5                                          
samba-tools                                                  x86_64                              4.20.2-2.el9_5                                     
samba-winbind                                             x86_64                              4.20.2-2.el9_5                                      
samba-winbind-modules                             x86_64                              4.20.2-2.el9_5               

after the update, this message appears:

Failed to enable unit: Unit file samba.service does not exist.
Failed to start samba.service: Unit samba.service not found.

**Comment by kozistan** [comment]
Should i provide more info of that update? Actually i needed to restore the snapshot because I'm running in on production so i would be in outage. I've try that twice and with the same.

Failed to enable unit: Unit file samba.service does not exist.
Failed to start samba.service: Unit samba.service not found.

**Comment by fumatchu** [comment]
It looks like the RPMS from Rocky are .el9_5? Did you upgrade this box to 9.5?

**Comment by fumatchu** [comment]
OK, For every major version this is a manual process where you must run the samba-dnf-pkg-update command and rebuild from mock. you have two options.  #1- build a new 9.5 server and re-run the Github installer to join this as another (new) DC server, replicate your Directory information, move your FSMO role, then remove 9.4 version server  from your environment. See here: https://wiki.samba.org/index.php/Demoting_a_Samba_AD_DC Do this in a TEST lab to make sure everything is smooth. The second option (again, I would do this in a TEST lab) running the command "samba-dnf-pkg-update" on your snapshot. This is in the github repo, but is also installed on your system. It invokes mock to rebuild your binaries as needed for the new version of samba being released. This is from the samba website, stating that if you installed from source, you must rebuild. https://wiki.samba.org/index.php/Updating_Samba "If you update Samba by compiling from the sources, you should be aware that code can be removed for various reasons. If code is removed and you compile and install Samba over your existing Samba installation, this can lead to old libs being left on disk, this can lead to errors. For this reason, it is recommended that you replace the entire Samba installation if you compile Samba from sources." (This is the suggestions above-#1) If you are OK being on 9.4, for now, I would suggest duplicating your environment in a test lab and trying these options. I have not tested 9.5 yet. I am guessing that everything should be fine.  I will try to do that this weekend. I'm sorry I know you wanted a simpler answer, but both of these options should work for you. Please let me know how things go

**Comment by kozistan** [comment]
So I've made snapshot and trying samba-dnf-pkg-update and this could be the issue, right?

`INFO: mock.py version 5.9 starting (python version = 3.9.18, NVR = mock-5.9-1.el9), args: /usr/libexec/mock/mock -r rocky-9-x86_64 --enablerepo=devel --define 'dist .el9_4.dc' --with dc samba-4.19.4-105.el9_4.dc.src.rpm
Start(bootstrap): init plugins
INFO: selinux enabled
Finish(bootstrap): init plugins
Start: init plugins
INFO: selinux enabled
Finish: init plugins
INFO: Signal handler active
Start: run
ERROR: Cannot find/open srpm: samba-4.19.4-105.el9_4.dc.src.rpm. Error: [Errno 2] No such file or directory: 'samba-4.19.4-105.el9_4.dc.src.rpm'
Directory walk started
Directory walk done - 65 packages
Temporary output repo path: /root/.samba/.repodata/`

**Comment by fumatchu** [comment]
If this is a snapshot and this is in your lab you should upgrade to 9.5 then run the command. Let me know what that returns

**Comment by kozistan** [comment]
so after upgrade to 9.5 and samba-dnf-pkg-update same output:

Failed to enable unit: Unit file samba.service does not exist.
Failed to start samba.service: Unit samba.service not found.

I've revert the changes and will wait for your tests. Thanks fumatchu, I really appreciate your help.

**Comment by fumatchu** [comment]
OK 9.5 totally broke RADS. Back to the drawing board..I will re view this over the next couple of days to get it working again and then try to upgrade it, but no promises on the upgrade component at this point in time.. sorry

**Comment by kozistan** [comment]
great news! looking forward

**Comment by kozistan** [comment]
> OK, I have a fresh install working again.. I will work on the updater over the weekend. Just wanted to give you a heads up :)

great news! looking forward

**Comment by kozistan** [comment]
just come into my mind, can i help you with anything? :)

**Comment by fumatchu** [comment]
Sorry, day job has me working on another project I need to get completed by tomorrow. I will get this completed by Frida. Thanks for your understanding

**Comment by fumatchu** [comment]
I'm sorry for the delay here.. I finally completed what I needed to at work...I will pull 9.4 tomorrow morning and start working through the upgrade to 9.5

**Comment by kozistan** [comment]
hi fumatchu, any update here?

**Comment by kozistan** [comment]
so i successfully installed samba on clean 9.5 without any issues as you have mentioned on the main page. For now I'm not transferring fsmo and sysvol to it, just made it secondary with running replication and will wait for this fix.

**Comment by fumatchu** [comment]
Hey, I'm sorry this took so long to do.. However I updated the script and tested from 9.1 through 9.5 for an in place upgrade. Things "seem" to be working on my end. I also put some validation in the script so that it stops before it makes any errors.  that being said the file name is the same. It's the samba-dnf-pkg-update file in the RADS repo. You can just copy the file on your system and run it from the root directory if you wish. It doesn't need to be named that. You can call it test or whatever you want. the crux of one of the issues seemed to be that a dnf update was not install the DC rpm after it's built. I got that taken care of. so this is a true upgrade All the config files stay in place. the server OS needs to be upgraded first to 9.5, from 9.4. After the system is upgraded to 9.5 you run samba-dnf-pkg-update (or whatever you want to call it). after the OS upgrade. after the updater does it's thing, you will be prompted to reboot. So keep in mind, samba will break after the 9.5 upgrade as we have seen, but this updater will correct the issues.

**Comment by kozistan** [comment]
Hi, this has worked out. All my samba AD controllers are up to date, it was smooth without any issues, thanks!
Closing this one.

**Comment by kozistan** [comment]
Hi, this has worked out. All my samba AD controllers are up to date, it was smooth without any issues, thanks!
Closing this one.

**Comment by fumatchu** [comment]
Just checking, you used the update script? 
If you did not I would also run the patch.. there were two other bugs I found that I corrected.  It's with the monitoring agent that detects when  a new version is available.. you can run it without having to compile again.. just run the patch, don't run the samba update script again. his is a patch script for any new bugs. It will display a dialog script and allow the End user to choose the patch to apply

#Current patches:

#Samba-dnf-pkgupdate- If Rocky was installed before January 11 2025, pkg update would not install dc rpm on upgrade. corrected.
#Please make sure you have manually set your DNS to something other than the Samba service before attempting to run this file
for example, use "nmtui" to edit the dns interface to 208.67.222.222 (OpenDNS), save and then run the commnad "systemctl restart NetworkManager", followed by nmcli to validate that the settings took effect.

If you receieve the error of:
Unable to save connection:           
802-3-ethernet.mac-address-denylist:
unknown property
(In nmtui)
you must "systemctl restart NetworkManager"
Modify the DNS entry to the DNS other than AD,  
then restart NetworkManager again before downloading the script 

Please keep in mind that this is also true if the script finds updates. YOU MAY run into this scenario above even if AD is operational and resolving
##END NOTE FOR SAMBA-DNF-PKG-UPDATE##

#(Just copy and paste the following line on the Rocky terminal or remote SSH terminal- remote suggested)
dnf -y install wget && cd /root &&  bash <(wget -qO- https://raw.githubusercontent.com/fumatchu/RADS/main/patch.sh)

**Comment by kozistan** [comment]
Yes I did the update with script samba-dnf-pkg-update file in the RADS repo. Anyway your patch has been applied now as you described. Everything seem to be working, thanks!

**Comment by fumatchu** [comment]
Awesome! Thank you! and thank you for your patience and time

**Comment by kozistan** [comment]
thanks for your great work fumatchu!
